src/basics.md

```diff
@@ -270,6 +271,54 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-regex-filter-log]: #ex-regex-filter-log
+<a name="ex-regex-filter-log"></a>
+## Filter a log file by matching multiple regular expressions
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Reads a file named `application.log` and only outputs the lines 
+containing “warning” or “error” (case insensitive). A [`regex::RegexSet`] is built
+with [`regex::RegexSetBuilder`] using the builder pattern.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+# use std::fs::File;
+# use std::io::{BufReader, BufRead};
+# 
+use regex::RegexSetBuilder;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Regex(regex::Error);
+#     }
+# }
+#
+fn run() -> Result<()> {
+    let log_path = "application.log";
+    let log_file = File::open(log_path)?;
+    let buffered = BufReader::new(log_file);
+
+    let set = RegexSetBuilder::new(&[r"warning", r"error"])
+                .case_insensitive(true)
+                .build()?;
+
+    for line in buffered.lines() {
+        let line = line?;
```

Comment:
in general shadowing is ok but i would suggest using iterator combinators (eg filter_map) to avoid bailing out on first error. Please see the [following example](https://brson.github.io/rust-cookbook/app.html#ex-tar-strip-prefix) for details.